airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -86,23 +86,10 @@ def __init__(
             component_factory=component_factory,
         )
 
+        # todo: We could remove state from initialization. Now that streams are grouped during the read(), a source
+        #  no longer needs to store the original incoming state. But maybe there's an edge case?
```

Comment:
added this comment as a todo, i don't think we use state anywhere, but in essence of fewer changes to fix the bug.